### PR TITLE
New CI and release workflow

### PR DESCRIPTION
**Pull request by omBratteng**
CI and build/deploy is now a separate workflow

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      -
+        name: Docker meta
+        id: docker_meta
+        uses: docker/metadata-action@v3.5.0
+        with:
+          images: |
+            ghcr.io/tus/tusd
+            tusproject/tusd
+          flavor: |
+            latest=${{ startsWith(github.ref, 'refs/tags/') }}
```
Comment (Acconut) [inline .github/workflows/release.yaml:30]: Does this set the `latest` tag for the current image?

File: Dockerfile [hunk 2]
```diff
@@ -1,37 +1,43 @@
-FROM golang:1.16-alpine AS builder
+FROM golang:1.17.2-alpine AS builder
```
Comment (Acconut) [inline Dockerfile:1]: Could you explain why you made these the changes in the Dockerfile? Do they enable multiarch builds, as you mentioned in https://github.com/tus/tusd/issues/511#issuecomment-945778847?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+          tags: |
+            type=sha
+            type=semver,pattern=v{{version}}
+            type=semver,pattern=v{{major}}.{{minor}}
+            type=semver,pattern=v{{major}}
```
Comment (Acconut) [inline .github/workflows/release.yaml:35]: Wow, that metadata action looks awesome :heart_eyes:

File: .github/workflows/release.yaml (hunk 1 above)
Comment (Acconut) [reply .github/workflows/release.yaml:30]: One more thing: We would like to only set the latest tag if the tag is no pre-release (i.e. not of the form `v1.2.3-pre1`). In that case, the tags v1 and v1.2 should also not be set. Is that easily possible or should we use an entirely different tag format for pre-releases?

File: .github/workflows/release.yaml (hunk 3 above)
Comment (omBratteng) [reply .github/workflows/release.yaml:35]: It is a pretty sweet action, does some great metadata magic

File: Dockerfile (hunk 2 above)
Comment (Acconut) [reply Dockerfile:1]: I was talking about all changes in the Dockerfile :) However, you second comment already explains what I wanted to know!

File: .github/workflows/release.yaml (hunk 1 above)
Comment (omBratteng) [reply .github/workflows/release.yaml]: If you check the runs I did, under the step Docker meta, and the Docker tags section, you see what tags it generates

`v300.0.0-pre1`
image

`v200.1.2`
image

`push to master` (in the test run it's to branch called `local-testing` for a quick test)
image

**Pull request merged**